app/src/pages/inside/projectSettingsPageContainer/projectSettingsPageContainer.jsx

```diff
@@ -14,39 +14,164 @@
+      type: PROJECT_SETTINGS_TAB_PAGE,
+      payload: { projectId, settingsTab: tabName },
+    }),
+    [projectId],
+  );
+
+  const createExtensionTabs = useCallback(() => {
+    return extensions.reduce(
+      (acc, extension) => ({
+        ...acc,
+        [extension.name]: {
+          name: extension.title || extension.name,
+          link: createTabLink(extension.name),
+          component: <ExtensionLoader extension={extension} />,
+          mobileDisabled: true,
+          eventInfo: SETTINGS_PAGE_EVENTS.extensionTabClick(extension.title || extension.name),
+        },
+      }),
+      {},
     );
-  }
-}
+  }, [createTabLink, extensions]);
+
+  const createItemsConfig = useCallback(() => {
+    const extensionTabs = createExtensionTabs();
+    const tabsConfig = {
+      [GENERAL]: {
+        name: formatMessage(messages.general),
+        link: createTabLink(GENERAL),
+        component: <GeneralTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.GENERAL_TAB,
+        mobileDisabled: true,
+      },
+      [INTEGRATIONS]: {
+        name: formatMessage(messages.integrations),
+        link: createTabLink(INTEGRATIONS),
+        component: <IntegrationsTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.INTEGRATIONS_TAB,
+      },
+      [NOTIFICATIONS]: {
+        name: formatMessage(messages.notifications),
+        link: createTabLink(NOTIFICATIONS),
+        component: <NotificationsTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.NOTIFICATIONS_TAB,
+        mobileDisabled: true,
+      },
+      [DEFECT]: {
+        name: formatMessage(messages.defect),
+        link: createTabLink(DEFECT),
+        component: <DefectTypesTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.DEFECT_TYPE_TAB,
+        mobileDisabled: true,
+      },
+      [ANALYSIS]: {
+        name: formatMessage(messages.analysis),
+        link: createTabLink(ANALYSIS),
+        component: <AutoAnalysisTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.AUTO_ANALYSIS_TAB,
+        mobileDisabled: true,
+      },
+      [PATTERN_ANALYSIS]: {
+        name: formatMessage(messages.patternAnalysis),
+        link: createTabLink(PATTERN_ANALYSIS),
+        component: <PatternAnalysisTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.PATTERN_ANALYSIS_TAB,
+        mobileDisabled: true,
+      },
+      [DEMO_DATA]: {
+        name: formatMessage(messages.demoData),
+        link: createTabLink(DEMO_DATA),
+        component: <DemoDataTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.DEMO_DATA_TAB,
+        mobileDisabled: true,
+      },
+    };
+    if (!canSeeDemoData(accountRole, userRole)) {
+      delete tabsConfig[DEMO_DATA];
+    }
+    Object.keys(extensionTabs).forEach((tab) => {
+      if (tabsConfig[tab]) {
+        tabsConfig[tab].component = extensionTabs[tab].component;
+
+        delete extensionTabs[tab];
+      }
+    });
+    return { ...tabsConfig, ...extensionTabs };
+  }, [accountRole, createExtensionTabs, createTabLink, formatMessage, userRole]);
```

Comment:
Looks like `formatMessage` can be omitted here as its change should not lead to itemsConfig recalculation.